CMakeLists.txt

```
@@ -150,9 +150,13 @@ endif()
 option(DEFAULT_FEATURE_STATE "Specify the default state for most optional features" True)
 mark_as_advanced(DEFAULT_FEATURE_STATE)
 
+# openssl/crypto
+set(ENABLE_OPENSSL True)
+set(ENABLE_WOLFSSL False)
```

My meaning here was that ENABLE_WOLFSSL should be an option, but ENABLE_OPENSSL should be set to true or false based on the value of that option. 

Framed a bit differently, _exactly_ one of these variables must be True. We can’t build if both are False (because we decided to have a hard dependency on a TLS implementation) or both are True (because that would break the abstraction layer). The original implementation you had here ensured that it wasn’t possible for both to be True, but not that it wasn’t possible for both to be False.

What I’m suggesting is essentially this:

```suggestion
option(ENABLE_WOLFSSL "Use WolfSSL in place of OpenSSL for TLS and crypto support" False)

if(ENABLE_WOLFSSL)
    set(ENABLE_OPENSSL False)
else()
    set(ENABLE_OPENSSL True)
endif()
```